### ISSUE TITLE
Other Gradle tasks are slower

### ISSUE THREAD

**Issue by mkjois** [DanySK/javadoc.io-linker#3]
Hi there! Thanks for this useful plugin.

I have a question about the implementation. Currently you try to download the package lists _at configure-time_ and warn about anything not on javadoc.io. Why is this? I've noticed it makes all other Gradle tasks slower if I have many dependencies. Why not do the following?

* Wrap this [options closure](https://github.com/DanySK/javadoc.io-linker/blob/master/src/main/groovy/org/danilopianini/gradle/javadociolinker/JavadocIOLinker.groovy#L39) in a `doFirst` closure, so the check is only done when the `javadoc` task is actually run
* Instead of performing the check by downloading the package lists, why not just check if it exists in the temp directory created by `downloadJavadocIOPackageLists`? The network call seems redundant.

I'd be happy to submit a PR, but I wanted to ask first. Thanks!

**Comment by DanySK** [comment]
Hi @mkjois 
sorry for the late reply. PR are extremely welcome, so please proceed!

Cheers

**Comment by DanySK** [comment]
Pull request merged, new version released. @mkjois, if you like to see your contribution recognized, I'd gladly update the build file including your personal details. Just provide me an entry in the form:
```groovy
                    contributor {
                        name 'Danilo Pianini'
                        email 'danilo.pianini@gmail.com'
                        url 'http://www.danilopianini.org/'
                    }

```

**Comment by mkjois** [comment]
@DanySK Sure! I don't have a website, but I suppose my Github will do
```
contributor {
    name 'Manny Jois'
    email 'm.k.jois@gmail.com'
    url 'https://github.com/mkjois'
}
```

**Comment by mkjois** [comment]
Fixed in #4

**Comment by DanySK** [comment]
You have been added to contributors, release 1.3 is out on Central. Thank you.
